### PR TITLE
Added Year View Widget.

### LINKED ISSUE #47: Add year view
Need to implement year view screen that shows user, a scrollable year calendar with mood colors.    
Add a text button on the top left of the  homePage and Navigate to year view screen.  

- [x] YearPage
- [x] Year Widget 
- [x] Navigation Route to YearPage 

You can look into this https://github.com/mennorenkens/flutter_scrolling_calendar package to get the idea of implementation.

### PR DESCRIPTION
**Pull request by cyber-venom003**
Added scrollable year view widget along with custom month tile. 
Fixes #47 .
Screenshot_1610071339

### CODE REVIEW

**Comment by cyber-venom003** [comment]
@hashirshoaeb Check the PR. Any changes are welcomed.

**Comment by hashirshoaeb** [comment]
Will review later tonight.

File: lib/widgets/year.dart [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            ),
+            child: buildYearMonths(context),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class MonthTile extends StatelessWidget {
```
Comment (hashirshoaeb) [inline lib/widgets/year.dart:97]: refactor the existing month widget and use that inplace of this.

File: lib/screens/year_view.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import 'package:flutter/cupertino.dart';
+import 'package:star_book/utils/date.dart';
+import 'package:star_book/utils/screen_sizes.dart';
+import 'package:star_book/widgets/year.dart';
+
```
Comment (hashirshoaeb) [inline lib/screens/year_view.dart:5]: file name: year_page.dart
class name YearPage

**Review by hashirshoaeb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**